cmd/spire-server/cli/agent/evict.go

```diff
@@ -0,0 +1,100 @@
+package agent
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	"github.com/spiffe/spire/pkg/common/idutil"
+	"github.com/spiffe/spire/proto/api/registration"
+
+	"golang.org/x/net/context"
+)
+
+//EvictConfig holds configuration for EvictCLI
+type EvictConfig struct {
+	// Socket path of registration API
+	RegistrationUDSPath string
+	// SpiffeID of the agent being evicted
+	SpiffeID string
+}
+
+// Validate will perform a basic validation on config fields
+func (c *EvictConfig) Validate() (err error) {
+	if c.RegistrationUDSPath == "" {
+		return errors.New("a socket path for registration api is required")
+	}
+
+	if c.SpiffeID == "" {
+		return errors.New("a SPIFFE ID is required")
+	}
+
+	// make sure SPIFFE ID is well formed
+	c.SpiffeID, err = idutil.NormalizeSpiffeID(c.SpiffeID, idutil.AllowAny())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//EvictCLI command for node "de-attestation"
+type EvictCLI struct {
+	RegistrationClient registration.RegistrationClient
+}
+
+func (EvictCLI) Synopsis() string {
+	return "De-attest an agent givent its spiffeID"
```

Comment:
Nit: IMO we should be consistent with the naming, maybe something like `Evict an agent given its SPIFFE ID`? Also, small typo s/givent/given/

Reply:
Agreed! Will standardize on "Evict"